server/src/main/java/org/opensearch/cluster/CompressionHelper.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.Version;
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.compress.CompressorFactory;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.common.io.stream.OutputStreamStreamOutput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+
+/**
+ * A helper class to utilize the compressed stream.
+ */
+public class CompressionHelper {
+    private static final Logger logger = LogManager.getLogger(CompressionHelper.class);
+
+    /**
+     * It'll always use compression before writing on a newly created output stream.
+     * @param writer Object which is going to write the content
+     * @param nodeVersion version of cluster node
+     * @param streamBooleanFlag flag used at receiver end to make intelligent decisions. For example, ClusterState
```

Comment:
This flag looks specific to ClusterState and doesnt seem generic and node version

Reply:
keep the method name as serializeClusterState and you can use this method from  for both full and diff cluster state. 

This class is compressing and serializing both. May be you name it specific to ClusterState

Reply:
These flags or fields we can write on stream even multiple times, then in same manner you read on receiver end write your logic around the values. My thought is to create a helper class which in future can be used by other communications also if anyone wants to use compression for a change or just want to use BytesTransportRequest (in that case another helper method can be added without compression). That's why I want to keep this class generic. 

My proposal : Let's keep the class generic, as method is specifically writing the bytes on stream specific for cluster state use case, I'll mark method name as serializeClusterState.

Also will move it to opensearch.common.compress so other code base can easily use it. Let me know if you're aware of a better sub package.